public/app/features/serviceaccounts/components/ServiceAccountsListItem.tsx

```diff
@@ -77,61 +83,98 @@ const ServiceAccountListItem = memo(
               aria-label="Role"
               value={serviceAccount.role}
-              disabled={!canUpdateRole}
+              disabled={!canUpdateRole || serviceAccount.isDisabled}
               onChange={(newRole) => onRoleChange(newRole, serviceAccount)}
             />
           </td>
         )}
-        <td className="link-td max-width-10">
-          <a
-            className="ellipsis"
-            href={editUrl}
-            title={getServiceAccountsEnabledStatus(serviceAccount.isDisabled)}
-            aria-label={getServiceAccountsAriaLabel(serviceAccount.name)}
-          >
-            {getServiceAccountsEnabledStatus(serviceAccount.isDisabled)}
-          </a>
-        </td>
         <td className="link-td max-width-10">
           <a
             className="ellipsis"
             href={editUrl}
             title="Tokens"
             aria-label={getServiceAccountsAriaLabel(serviceAccount.name)}
           >
-            <span>
-              <Icon name={'key-skeleton-alt'}></Icon>
-            </span>
-            &nbsp;
-            {serviceAccount.tokens}
+            <div className={cx(styles.tokensInfo, { [styles.tokensInfoSecondary]: !serviceAccount.tokens })}>
+              <span>
+                <Icon name="key-skeleton-alt"></Icon>
+              </span>
+              {serviceAccount.tokens || 'No tokens'}
+            </div>
           </a>
         </td>
-        {contextSrv.hasPermissionInMetadata(AccessControlAction.ServiceAccountsDelete, serviceAccount) && (
-          <td>
-            <Button
-              size="sm"
-              variant="destructive"
-              onClick={() => {
-                onSetToRemove(serviceAccount);
-              }}
-              icon="times"
-              aria-label="Delete service account"
-            />
-          </td>
-        )}
+        <td>
+          <HorizontalGroup justify="flex-end">
+            {contextSrv.hasPermission(AccessControlAction.ServiceAccountsWrite) && !serviceAccount.tokens && (
+              <Button onClick={() => onAddTokenClick(serviceAccount)} disabled={serviceAccount.isDisabled}>
+                Add token
+              </Button>
+            )}
+            {contextSrv.hasPermissionInMetadata(AccessControlAction.ServiceAccountsWrite, serviceAccount) &&
+              (serviceAccount.isDisabled ? (
+                <Button variant="primary" onClick={() => onEnable(serviceAccount)}>
+                  Enable
+                </Button>
+              ) : (
+                <Button variant="secondary" onClick={() => onDisable(serviceAccount)}>
+                  Disable
+                </Button>
+              ))}
+            {contextSrv.hasPermissionInMetadata(AccessControlAction.ServiceAccountsDelete, serviceAccount) && (
+              <IconButton
+                className={styles.deleteButton}
+                name="trash-alt"
+                size="sm"
+                onClick={() => onRemoveButtonClick(serviceAccount)}
```

Comment:
```suggestion
                onClick={() => onRemoveButtonClick(serviceAccount)}
                aria-label={`Delete service account ${serviceAccount.name}`}
```
This will also make it easier to write tests since we can now query the button by the label.